cpu/sam0_common/include/periph_cpu_common.h

```diff
@@ -945,6 +945,18 @@ typedef struct {
  */
 #define WDT_HAS_INIT                   (1)
 
+/**
+ * @brief   Frequency meter configuration
+ *
+ */
+#ifdef MODULE_PERIPH_FREQM
+struct freqm_config {
+    gpio_t msr_clk_src;     /*< GPIO at which the frequency is to be measured */
+    uint8_t ref_clk_src;    /*< GCLK source select for reference */
+    uint8_t ref_clk_cnt;    /*< Measurement duration in number of reference clock cycles */
```

Comment:
This sounds like something that could also be a parameter to `freqm_frequency_get()` (but let's use µs there, not clock cycles. We can always calculate the clock cycles with `((uint64_t)period_us * sam0_gclk_freq(gclk_src)) / US_PER_SEC`)

Reply:
Hardwarewise the measurement duration is constrained to 255 reference clock cycles. So I am wondering how to fail or if it should fail if the user sets a duration that would exceed this value. 

A possibility would be to run several measurements automatically one after the other and then average the result. However, you would then have to split the selected duration evenly over several measurements, I guess. Or should you leave this to the user and only allow a maximum of 255 cycles?

Reply:
> should you leave this to the user and only allow a maximum of 255 cycles?

Yes, let's keep it simple.
I would suggest to just clip it, e.g. if the value goes above 255 cycles, limit it to 255 cycles.
If it drops below 1 cycle, set it to 1 instead.

That way a user of the API doesn't have to care too much about the underlying hardware and will still get acceptable results.